parquet/api.py

```diff
@@ -0,0 +1,280 @@
+"""parquet - read parquet files."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+import os
+import pandas as pd
+import re
+import struct
+import thriftpy
+
+from .core import ParquetException, read_thrift
+from .thrift_structures import parquet_thrift
+from .writer import write
+from . import core, schema, converted_types, encoding
+
+
+def def_open(f):
+    return open(f, 'rb')
+
+
+class ParquetFile(object):
+    """For now: metadata representation
+
+    Parameters
+    ----------
+    fname: path/URL string
+    verify: test file start/end bytes
+    open_with: function returning an open file
+    """
+
+    def __init__(self, fname, verify=False, open_with=def_open):
+        if isinstance(fname, str):
+            try:
+                # backend may not have something equivalent to `isdir()`
+                f = open_with(fname)
+            except (IOError, OSError):
+                fname = os.path.join(fname, '_metadata')
```

Comment:
Does the user or backend file-system then need to provide the separation character?